tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CaffeineCacheStats.java

```diff
@@ -37,19 +41,39 @@ private CaffeineCacheStats() {
      * @param cache cache to instrument
      * @param name cache name
      */
-    public static <C extends Cache<?, ?>> void registerCache(MetricRegistry registry, Cache<?, ?> cache, String name) {
-        registerCache(registry, cache, name, Clock.defaultClock());
+    public static void registerCache(MetricRegistry registry, Cache<?, ?> cache, String name) {
+        checkNotNull(registry, "registry");
+        checkNotNull(cache, "cache");
+        checkNotNull(name, "name");
+
+        CaffeineCacheMetrics.create(cache, name)
+                .getMetrics()
+                .forEach((key, value) -> MetricRegistries.registerWithReplacement(registry, key, value));
     }
 
-    @VisibleForTesting
-    static void registerCache(MetricRegistry registry, Cache<?, ?> cache, String name, @VisibleForTesting Clock clock) {
+    /**
+     * Register specified cache with the given metric registry.
+     *
+     * @param registry metric registry
+     * @param cache cache to instrument
+     * @param name cache name
+     */
+    public static void registerCache(TaggedMetricRegistry registry, Cache<?, ?> cache, SafeArg<String> name) {
         checkNotNull(registry, "registry");
         checkNotNull(cache, "cache");
         checkNotNull(name, "name");
-        checkNotNull(clock, "clock");
-        CaffeineCacheMetricSet.create(cache, name, clock)
+
+        CaffeineCacheTaggedMetrics.create(cache, name)
                 .getMetrics()
-                .forEach((key, value) -> MetricRegistries.registerWithReplacement(registry, key, value));
+                .forEach((metricName, gauge) -> {
+                    registry.remove(metricName);
+                    registry.gauge(metricName, gauge);
+                });
     }
 
+    static <K extends Comparable<K>> Map<K, Gauge<?>> createCacheGauges(
+            Cache<?, ?> cache,
+            Function<String, K> metricNamer) {
+        return CacheMetrics.createMetrics(CaffeineStats.create(cache, 1, TimeUnit.SECONDS), metricNamer);
```

Comment:
Was the change from 5 seconds to 1 second intentional?

Reply:
Looks like some caches are moving from .5s -> 1s, and others from 5s -> 1s. Using 1 second across the board is easier for me to follow.

Reply:
yeah, with the switch to Guava memoized supplier with expiration, there's not a good way to inject a `clock` for testing, so the the 5 second caching was excessive and slows down tests, and allows for cases where we want 1 second frequency stats.